File: Universidad/Ciclo 6/Programacion 3/Laboratorio7/Java/Benchele/BenchelePersistance/src/main/java/com/mycompany/dao/AlmacenDAO.java
package com.mycompany.dao;

import java.util.ArrayList;
import com.mycompany.benchelemodel.AlmacenesDTO;

public interface AlmacenDAO {
    
    public Integer insertar(AlmacenesDTO almacen);
    
    public AlmacenesDTO obtenerPorId(Integer almacenId);
    
    public ArrayList<AlmacenesDTO> listarTodos();
    
    public Integer modificar(AlmacenesDTO almacen);
    
    public Integer eliminar(AlmacenesDTO almacen);
    
    public ArrayList<AlmacenesDTO> listarTodosPorProcedure();
}

File: Universidad/Ciclo 6/Programacion 3/Laboratorio7/CSharp/Benchele/BenchelePersistance/DAO/AlmacenDAO.cs
using BencheleModel;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BenchelePersistance.DAO
{
    public interface AlmacenDAO
    {
        int Insertar(AlmacenesDTO almacen);

        AlmacenesDTO ObtenerPorId(int almacenId);

        IList<AlmacenesDTO> ListarTodos();

        int Modificar(AlmacenesDTO almacen);

        int Eliminar(AlmacenesDTO almacen);

        //Nuevo hecho por mi

        IList<AlmacenesDTO> ListarTodosPorProcedure();
        IList<AlmacenesDTO> ListarPorCriterio(string criterio);
    }
}
